altair/utils/schemapi.py

```diff
@@ -564,11 +564,17 @@ def to_dict(self, validate=True, ignore=None, context=None):
             try:
                 self.validate(result)
             except jsonschema.ValidationError as err:
-                raise SchemaValidationError(self, err)
+                raise SchemaValidationError(self, err) from err
```

Comment:
```suggestion
                # We do not raise `from err` and therfore ignore 
                # the B904 linting rule as else the resulting
                # traceback is very long as it contains part
                # of the Vega-Lite schema. It would also first
                # show the less helpful ValidationError instead of
                # the more user friendly SchemaValidationError
                raise SchemaValidationError(self, err)  # noqa: B904
```

Reply:
I've modified your suggestion a little bit to `raise from None` instead of `raise # noqa: B209`. I'm not entirely sure if that leads to the same behavior. Do you have an example how you checked this?

Reply:
Double-checked and that works as well. It gives the same traceback as in master which is good. I tested it with:

```python
import altair as alt
from vega_datasets import data

(
    alt.Chart(data.barley())
    .mark_bar()
    .encode(
        x=alt.X("variety", unknown=2),
        y=alt.Y("sum(yield)", stack="asdf"),
    )
)
```